src/otx/core/model/instance_segmentation.py

```diff
@@ -482,8 +484,8 @@ def _forward_explain_inst_seg(
             # Export case, consists of tensors
             # For OV task saliency map are generated on MAPI side
             saliency_map = torch.empty(1, dtype=torch.uint8)
-        elif isinstance(predictions, list) and isinstance(predictions[0], InstanceData):
-            # Predict case, consists of InstanceData
+        elif isinstance(predictions, list) and isinstance(predictions[0], (InstanceData, dict)):
+            # Predict case, consists of InstanceData or dict
```

Comment:
Is `self.explain_fn` on the next line ready for a dict?

Reply:
yeah, `self.explain_fn = InstSegExplainAlgo.func`  supports list of dictionaries.

```py
def func(
        self,
        predictions: list[InstanceData] | list[dict[str, Tensor]],
        _: int = -1,
    ) -> list[np.array]:
        """Generate saliency maps from predicted masks by averaging and normalizing them per-class.

        Args:
            predictions (list[InstanceData] | list[dict[str, Tensor]): Predictions of Instance Segmentation model.

        Returns:
            torch.Tensor: Class-wise Saliency Maps. One saliency map per each class - [batch, class_id, H, W]
        """
        batch_saliency_maps = []
        for prediction in predictions:
            class_averaged_masks = self.average_and_normalize(prediction, self.num_classes)
            batch_saliency_maps.append(class_averaged_masks)
        return torch.stack(batch_saliency_maps)

@classmethod
    def average_and_normalize(
        cls,
        pred: InstanceData | dict[str, torch.Tensor],
        num_classes: int,
    ) -> np.array:

```